kafka.go

```diff
@@ -168,12 +168,7 @@ func (kafka *kafkaReader) Seek(ctx context.Context, offset int64) (int64, error)
 		kafka.closeAsync()
 	}
 
-	broker, err := kafka.client.Leader(kafka.topic, kafka.partition)
-	if err != nil {
-		return 0, err
-	}
-
-	offset, err = kafka.getOffset(broker, offset)
+	offset, err := kafka.getOffset(ctx, offset)
```

Comment:
I'm confused about why we need to call getOffset in this method, why not have the cursor initially be -1 (or -2), then have the goroutine fetch the offset if it needs to, then rewrite Read to do something like this:
```go
func (r *kafkaReader) Read(ctx context.Context) (msg Message, err error) {
  // 1. receive the message
  // 2. update the offset
}
```